terraform/provider.tf

```diff
@@ -0,0 +1,18 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 2.26"
+    }
+  }
+
+  backend "azurerm" {
+    storage_account_name = "oxterraformbackend"
```

Comment:
extract to parameter

Reply:
variables are not allowed in this section... I will need to see later how I make it all part of tf. (want the storage it self to be created as well)